polymorphic_tree/tests/test_models.py

```diff
@@ -0,0 +1,126 @@
+
+        # from a manual list
+        objects = Model2A.objects.get_real_instances(list(qs))
+        self.assertEqual(repr(objects[0]), '<Model2A: id 1, parent (PolymorphicTreeForeignKey), field1 (CharField), lft (PositiveIntegerField), rght (PositiveIntegerField), tree_id (PositiveIntegerField), level (PositiveIntegerField)>')
+        self.assertEqual(repr(objects[1]), '<Model2B: id 2, parent (PolymorphicTreeForeignKey), field1 (CharField), lft (PositiveIntegerField), rght (PositiveIntegerField), tree_id (PositiveIntegerField), level (PositiveIntegerField), field2 (CharField)>')
+        self.assertEqual(repr(objects[2]), '<Model2C: id 3, parent (PolymorphicTreeForeignKey), field1 (CharField), lft (PositiveIntegerField), rght (PositiveIntegerField), tree_id (PositiveIntegerField), level (PositiveIntegerField), field2 (CharField), field3 (CharField)>')
+        self.assertEqual(repr(objects[3]), '<Model2D: id 4, parent (PolymorphicTreeForeignKey), field1 (CharField), lft (PositiveIntegerField), rght (PositiveIntegerField), tree_id (PositiveIntegerField), level (PositiveIntegerField), field2 (CharField), field3 (CharField), field4 (CharField)>')
+
+    def test_translate_polymorphic_q_object(self):
+        self.create_model2abcd()
+
+        q = Model2A.translate_polymorphic_Q_object(Q(instance_of=Model2C))
+        objects = Model2A.objects.filter(q)
+        self.assertEqual(repr(objects[0]), '<Model2C: id 3, parent (PolymorphicTreeForeignKey), field1 (CharField), lft (PositiveIntegerField), rght (PositiveIntegerField), tree_id (PositiveIntegerField), level (PositiveIntegerField), field2 (CharField), field3 (CharField)>')
+        self.assertEqual(repr(objects[1]), '<Model2D: id 4, parent (PolymorphicTreeForeignKey), field1 (CharField), lft (PositiveIntegerField), rght (PositiveIntegerField), tree_id (PositiveIntegerField), level (PositiveIntegerField), field2 (CharField), field3 (CharField), field4 (CharField)>')
+
+    def test_base_manager(self):
+        def show_base_manager(model):
+            return "{0} {1}".format(
+                repr(type(model._base_manager)),
+                repr(model._base_manager.model)
+            )
+
+        self.assertEqual(show_base_manager(PlainA), "<class 'django.db.models.manager.Manager'> <class 'polymorphic_tree.tests.test_models.PlainA'>")
+        self.assertEqual(show_base_manager(PlainB), "<class 'django.db.models.manager.Manager'> <class 'polymorphic_tree.tests.test_models.PlainB'>")
+        self.assertEqual(show_base_manager(PlainC), "<class 'django.db.models.manager.Manager'> <class 'polymorphic_tree.tests.test_models.PlainC'>")
+
+        self.assertEqual(show_base_manager(Model2A), "<class 'polymorphic_tree.managers.PolymorphicMPTTModelManager'> <class 'polymorphic_tree.tests.test_models.Model2A'>")
+        self.assertEqual(show_base_manager(Model2B), "<class 'django.db.models.manager.Manager'> <class 'polymorphic_tree.tests.test_models.Model2B'>")
```

Comment:
I'm not sure here. TreeManager maybe if you following the "manager - polymorphic" rule, maybe `PolymorphicMPTTModelManager` because the functionality needs to be there.

I guess it depends on the effect that `_base_manager` has.